# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build ncurses-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal -I m4; autoconf; autoheader -f; automake -a -c
./configure --disable-static --disable-kill --disable-pidof --sbindir=/bin --prefix=

# build
make

# install
install -d $DESTDIR/bin
install pwdx $DESTDIR/bin
install watch $DESTDIR/bin
install .libs/free $DESTDIR/bin
install .libs/pgrep $DESTDIR/bin
ln -sf pgrep $DESTDIR/bin/pkill
install .libs/pmap $DESTDIR/bin
install .libs/slabtop $DESTDIR/bin
install .libs/sysctl $DESTDIR/bin
install .libs/tload $DESTDIR/bin
install .libs/uptime $DESTDIR/bin
install .libs/vmstat $DESTDIR/bin
install .libs/w $DESTDIR/bin
install ps/.libs/pscommand $DESTDIR/bin/ps
install top/.libs/top $DESTDIR/bin
install -d $DESTDIR/include/proc
install -m 644 proc/procps.h $DESTDIR/include/proc
install -m 644 proc/pwcache.h $DESTDIR/include/proc
install -m 644 proc/readproc.h $DESTDIR/include/proc
install -d $DESTDIR/lib
install proc/.libs/libprocps.so.8.* $DESTDIR/lib/libprocps.so.8
ln -sf libprocps.so.8 $DESTDIR/lib/libprocps.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 proc/libprocps.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/procps-ng-build.pc << EOF
Name: procps-ng
Version: 3.3.16
Description: utilities and library to monitor the system and its processes
Requires: $REQUIRES

devel=\\
/include/proc/procps.h \\
/include/proc/pwcache.h \\
/include/proc/readproc.h \\
/lib/libprocps.so \\
/lib/pkgconfig/libprocps.pc

exec=\\
/bin/free \\
/bin/pgrep \\
/bin/pkill \\
/bin/pmap \\
/bin/ps \\
/bin/pwdx \\
/bin/slabtop \\
/bin/sysctl \\
/bin/tload \\
/bin/top \\
/bin/uptime \\
/bin/vmstat \\
/bin/w \\
/bin/watch \\
/lib/libprocps.so.8
EOF

exit
--- procps-ng-3.3.16.orig/Makefile.am
+++ procps-ng-3.3.16/Makefile.am
@@ -41,7 +41,6 @@
 	ps/pscommand \
 	free \
 	pgrep \
-	pkill \
 	pmap \
 	pwdx \
 	tload \
@@ -53,7 +52,6 @@
 	ps/pscommand \
 	free \
 	pgrep \
-	pkill \
 	pmap \
 	uptime \
 	vmstat \
@@ -190,7 +188,6 @@
 
 free_SOURCES = free.c lib/strutils.c lib/fileutils.c
 pgrep_SOURCES = pgrep.c lib/fileutils.c lib/nsutils.c
-pkill_SOURCES = pgrep.c lib/fileutils.c lib/nsutils.c
 pmap_SOURCES = pmap.c lib/fileutils.c
 if !CYGWIN
 pwdx_SOURCES = pwdx.c lib/fileutils.c
@@ -325,7 +322,6 @@
 	kill_LDADD = $(CYGWINFLAGS)
 	free_LDADD = $(CYGWINFLAGS)
 	pgrep_LDADD = $(CYGWINFLAGS)
-	pkill_LDADD = $(CYGWINFLAGS)
 	pidof_LDADD = $(CYGWINFLAGS)
 	pmap_LDADD = $(CYGWINFLAGS)
 	snice_LDADD = $(CYGWINFLAGS)
@@ -353,3 +349,8 @@
 	echo "Getting the latest translations from translationproject.org..."
 	rsync -Lrtvz  translationproject.org::tp/latest/procps-ng/  po
 	rsync -Lrtvz  translationproject.org::tp/latest/procps-ng-man/  man-po
+
+install-exec-hook:
+	cd $(DESTDIR)$(bindir) && \
+	rm -f pkill$(EXEEXT) && \
+	$(LN_S) pgrep$(EXEEXT) pkill$(EXEEXT)
--- procps-ng-3.3.16.orig/m4/codeset.m4
+++ procps-ng-3.3.16/m4/codeset.m4
@@ -0,0 +1,23 @@
+# codeset.m4 serial AM1 (gettext-0.10.40)
+dnl Copyright (C) 2000-2002 Free Software Foundation, Inc.
+dnl This file is free software, distributed under the terms of the GNU
+dnl General Public License.  As a special exception to the GNU General
+dnl Public License, this file may be distributed as part of a program
+dnl that contains a configuration script generated by Autoconf, under
+dnl the same distribution terms as the rest of that program.
+
+dnl From Bruno Haible.
+
+AC_DEFUN([AM_LANGINFO_CODESET],
+[
+  AC_CACHE_CHECK([for nl_langinfo and CODESET], am_cv_langinfo_codeset,
+    [AC_TRY_LINK([#include <langinfo.h>],
+      [char* cs = nl_langinfo(CODESET);],
+      am_cv_langinfo_codeset=yes,
+      am_cv_langinfo_codeset=no)
+    ])
+  if test $am_cv_langinfo_codeset = yes; then
+    AC_DEFINE(HAVE_LANGINFO_CODESET, 1,
+      [Define if you have <langinfo.h> and nl_langinfo(CODESET).])
+  fi
+])
--- procps-ng-3.3.16.orig/m4/lcmessage.m4
+++ procps-ng-3.3.16/m4/lcmessage.m4
@@ -0,0 +1,32 @@
+# lcmessage.m4 serial 3 (gettext-0.11.3)
+dnl Copyright (C) 1995-2002 Free Software Foundation, Inc.
+dnl This file is free software, distributed under the terms of the GNU
+dnl General Public License.  As a special exception to the GNU General
+dnl Public License, this file may be distributed as part of a program
+dnl that contains a configuration script generated by Autoconf, under
+dnl the same distribution terms as the rest of that program.
+dnl
+dnl This file can can be used in projects which are not available under
+dnl the GNU General Public License or the GNU Library General Public
+dnl License but which still want to provide support for the GNU gettext
+dnl functionality.
+dnl Please note that the actual code of the GNU gettext library is covered
+dnl by the GNU Library General Public License, and the rest of the GNU
+dnl gettext package package is covered by the GNU General Public License.
+dnl They are *not* in the public domain.
+
+dnl Authors:
+dnl   Ulrich Drepper <drepper@cygnus.com>, 1995.
+
+# Check whether LC_MESSAGES is available in <locale.h>.
+
+AC_DEFUN([AM_LC_MESSAGES],
+[
+  AC_CACHE_CHECK([for LC_MESSAGES], am_cv_val_LC_MESSAGES,
+    [AC_TRY_LINK([#include <locale.h>], [return LC_MESSAGES],
+       am_cv_val_LC_MESSAGES=yes, am_cv_val_LC_MESSAGES=no)])
+  if test $am_cv_val_LC_MESSAGES = yes; then
+    AC_DEFINE(HAVE_LC_MESSAGES, 1,
+      [Define if your <locale.h> file defines LC_MESSAGES.])
+  fi
+])
